src/lib.rs

```diff
@@ -428,11 +428,11 @@ impl<K: EnrKey> Enr<K> {
     #[must_use]
     pub fn verify(&self) -> bool {
         let pubkey = self.public_key();
-        match self.id() {
-            Some(ref id) if id == "v4" => pubkey.verify_v4(&self.rlp_content(), &self.signature),
-            // unsupported identity schemes
-            _ => false,
-        }
```

Comment:
It looks to me like the previous code would be more extensible for new identity schemes and should probably be kept this way

Reply:
For this maybe we should remove the nursery level lints or allow it. Open to opinions

Reply:
I see, I don't have strong opinions on lint levels and have not used nursery lints before, so I'll just allow the lint instead of removing all nursery lints